src/API/QueryFieldServiceInterface.php

```diff
@@ -24,16 +26,40 @@ interface QueryFieldServiceInterface
     public function loadContentItems(Content $content, string $fieldDefinitionIdentifier): iterable;
 
     /**
-     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * Counts the total results of a query.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Content|\eZ\Publish\API\Repository\Values\Content\Location $content
      * @param string $fieldDefinitionIdentifier
      *
-     * @return \eZ\Publish\API\Repository\Values\Content\Content[]
+     * @return int
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
     public function countContentItems(Content $content, string $fieldDefinitionIdentifier): int;
 
+    /**
+     * Executes a paginated query and return the requested content items slice.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Content|\eZ\Publish\API\Repository\Values\Content\Location $content
+     * @param string $fieldDefinitionIdentifier
+     * @param int $offset
+     * @param int $limit
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Content[]
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
     public function loadContentItemsSlice(Content $content, string $fieldDefinitionIdentifier, int $offset, int $limit): iterable;
 
-    public function getPaginationConfiguration(Content $content, string $fieldDefinitionIdentifier): int;
+    /**
+     * Returns the configured items per page for a query field definition, or false if pagination is disabled.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Content|\eZ\Publish\API\Repository\Values\Content\Location $content
+     * @param string $fieldDefinitionIdentifier
+     *
+     * @return int|false
```

Comment:
warning: if this is true, it will result in a fatal error when returning `false`

Reply:
Damn, you spotted it. I wasn't sure what to do with that inconsistency 🤣

Reply:
>> warning: if this is true, it will result in a fatal error when returning `false`
>
> Damn, you spotted it. I wasn't sure what to do with that inconsistency

The fact remains that you cannot keep it this way, sorry.

Reply:
OK. We could change it so that it returns 0 if pagination is disabled. Since a page limit of 0 doesn't make any sense (and should actually be rejected by validation), it would work while respecting the return hint. Would that be okay ?

Reply:
What was the behavior until now?

Reply:
As you wish guys. I have prepared a change where it will return 0 when pagination is disabled, and I think it's fine, but in the end it's your code :)

Reply:
Currently fixed by returning 0 for disabled or the page limit if enabled. Let me know if it works for you.

Reply:
And added test for both enabled and disabled cases.